dbms/src/Storages/ColumnsDescription.cpp

```diff
@@ -75,22 +75,32 @@ String ColumnsDescription::toString() const
     {
         for (const auto & column : columns)
         {
-            const auto it = defaults.find(column.name);
+            const auto defaults_it = defaults.find(column.name);
+            const auto comments_it = comments.find(column.name);
 
             writeBackQuotedString(column.name, buf);
             writeChar(' ', buf);
             writeText(column.type->getName(), buf);
-            if (it == std::end(defaults))
+
+            const bool exist_comment = comments_it != std::end(comments) && !comments_it->second.expression;
+            if (defaults_it != std::end(defaults))
+            {
+                writeChar('\t', buf);
+                writeText(DB::toString(defaults_it->second.kind), buf);
+                writeChar('\t', buf);
+                writeText(queryToString(defaults_it->second.expression), buf);
+            }
+            else if (exist_comment)
             {
-                writeChar('\n', buf);
-                continue;
+                writeChar('\t', buf);
             }
-            else
+
+            if (exist_comment)
+            {
                 writeChar('\t', buf);
+                writeText(queryToString(comments_it->second.expression), buf);
```

Comment:
Somehow comments are not serialized in this code. You can check this by creating a Replicated table - the `<path_to_table>/columns` node doesn't contain column comments and the table creation fails with the `Table structure in ZooKeeper is too different from local table structure.` error.